tests/unit/basic_crawler/test_basic_crawler.py

```diff
@@ -78,7 +99,9 @@ async def handler(context: BasicCrawlingContext) -> None:
 
 
 async def test_retries_failed_requests() -> None:
-    crawler = BasicCrawler(request_provider=RequestList(['http://a.com/', 'http://b.com/', 'http://c.com/']))
+    request_queue = await RequestQueue.open()
+    await request_queue.add_requests_batched(['http://a.com/', 'http://b.com/', 'http://c.com/'])
+    crawler = BasicCrawler(request_provider=request_queue)
```

Comment:
Is it important for these tests to use explicit request provider? Can't we use just pass the requests to the `run`?

Reply:
It is important in one or two of those tests, I believe. I'll see if I can change it in the other cases.